Remove fastcrypto: take 2

ljedrz opened the pull request:
Applies on top of `fix/integration` and supersedes https://github.com/eqlabs/bullshark-bft/pull/37. Compiles, but haven't run the tests yet. I moved the changes from the former PR and applied some new ones where applicable.

Also, there's a change I didn't understand in the former PR - the `narwhal.yaml` one, so be extra vigilant about that bit.

ljedrz committed:
feat: switch to ed25519_dalek derivation and remove fastcrypto

Co-authored-by: Jos Dehaes <jos.dehaes@gmail.com>
Signed-off-by: ljedrz <ljedrz@gmail.com>

ljedrz committed:
chore: adjust the lockfile

Signed-off-by: ljedrz <ljedrz@gmail.com>

niklaslong commented on narwhal/crypto/src/network_keypair.rs:
```diff
@@ -0,0 +1,108 @@
+        Self(ed25519::Keypair::from_bytes(&self.0.to_bytes()).unwrap())
+    }
+
+    pub fn encode_base64(&self) -> String {
```
There's an `EncodeDecodeBase64` trait in the module, perhaps these could be moved to an impl of that. We did it initially to preserve some of the key APIs when changing the crypto.

niklaslong commented on narwhal/node/tests/staged/narwhal.yaml:
```diff
@@ -85,10 +85,7 @@ WorkerIndex:
-        TUPLEARRAY:
-          CONTENT: U8
-          SIZE: 32
+    - name: STR
```
I think this is expected, the underlying structure for the `NetworkPublicKey`, which is the `name` field of the `WorkerInfo` struct changed. @joske might have more insights to provide.

joske replied:
yes this was already in the original PR

niklaslong commented:
I've managed to reproduce the CI failure locally; not sure what's up with `test_determinism`.

joske reviewed (approved):
I wanted to do this tonight, but you already did :-D

niklaslong committed:
test: fix non-determinism test

This commit loosens the assertions to no longer include certificate
fields that contain a timestamp.

niklaslong committed:
clippy: misc lints

joske committed:
fix: move encode/decode into trait impl

joske committed:
fix: doctests
